templates/webhook/new.tmpl

```diff
@@ -1,5 +1,13 @@
 <h4 class="ui top attached header">
-	{{if .PageIsSettingsHooksNew}}{{.locale.Tr "repo.settings.add_webhook"}}{{else}}{{.locale.Tr "repo.settings.update_webhook"}}{{end}}
+	{{if .CustomHeaderTitle}}
+		{{.CustomHeaderTitle}}
+	{{else}}
+		{{if .PageIsSettingsHooksNew}}
+			{{.locale.Tr "repo.settings.add_webhook"}}
+		{{else}}
+			{{.locale.Tr "repo.settings.update_webhook"}}
+		{{end}}
```

Comment:
You could make every caller pass the `CustomHeaderTitle` to get rid of such fragile `if` blocks.